app/src/main/java/edu/usf/sas/pal/muser/util/EventUtils.kt

```diff
@@ -0,0 +1,99 @@
+package edu.usf.sas.pal.muser.util
+
+import android.content.Context
+import android.util.Log
+import com.simplecity.amp_library.model.Album
+import com.simplecity.amp_library.model.AlbumArtist
+import com.simplecity.amp_library.model.Song
+import com.simplecity.amp_library.utils.MusicServiceConnectionUtils
+import edu.usf.sas.pal.muser.model.*
+
+object EventUtils {
+    /**
+     * Function to populate the PlayerEvent data class.
+     * @param song - The song for which the event occurred.
+     * @param capturedEvent - The event that was captured.
+     * @param context - The context of the Fragment.
+     * @return PlayerEvent object
+     */
+    @JvmStatic
+    fun newPlayerEvent(song: Song?, capturedEvent: PlayerEventType?, context: Context?): PlayerEvent {
```

Comment:
Let's try to remove the `?` from `PlayerEventType` and `Context` - I believe `Song` should be the only nullable parameter here for player events that don't have songs. (is that right, or are we assuming that all player events have songs now?)

Reply:
That should also get rid of the `!!` below. We want to avoid those.

Reply:
I think it's safe to assume that `song` is always populated since we're logging at the lowest level of the service as possible. But, if we wanna add an extra layer of protection just in case, then I can go ahead and remove the `?` and add null checks.

Reply:
Looking at the code, if I change to this (removing all `?` and `!!`) it builds and runs:

```kotlin
    @JvmStatic
    fun newPlayerEvent(song: Song, capturedEvent: PlayerEventType, context: Context): PlayerEvent {
        val currentTimeMS = System.currentTimeMillis()
        val nanoTime = System.nanoTime()
        val songData = SongData(song, context)
        val seekPositionMs = position
        return PlayerEvent(capturedEvent, currentTimeMS, nanoTime, seekPositionMs,
                songData)
    }
```

Let's do that. `song` can potentially be `null` (looking at the Java classes it's called from), but I think we actually want to crash in this case instead of not logging the event so that we know we are missing events via crash logs in Google Play. As long as there isn't a valid use case where we expect `song` to be `null` - it sounds like there isn't.